pkg/agent/secondarynetwork/init_linux.go

```diff
@@ -43,30 +43,56 @@ var (
 	newOVSBridgeFn    = ovsconfig.NewOVSBridge
 )
 
-// Initialize sets up OVS bridges and starts the Pod controller for secondary networks.
-func Initialize(
+type Controller struct {
+	ovsBridgeClient ovsconfig.OVSBridgeClient
+	secNetConfig    *agentconfig.SecondaryNetworkConfig
+	podController   *podwatch.PodController
+}
+
+func NewController(
```

Comment:
Could we move this one to init.go? It seems have no OS specific code?

Reply:
According to the original logic, Windows does not supported SecondaryNetwork feature, so we would expect to return error on Windows when calling function `NewController`. If we move to `init.go`, we can't satisfy this precheck on Windows. 

I can move the struct definition of Controller to `init.go`, and prefer to leave different functions of `NewController` in init_linux.go and init_windows.go. 

What is your thought @jianjuns ?

Reply:
update: I tried to move the struct of `Controller` to init.go, then I would hit golangci-lint failures, because the properties of `ovsBridgeClient` and `secNetConfig` are not used in init.go. So I moved it back.

Reply:
If we want not Linux feature code to be in common files, then there are many more code / funcs to move to _linux.go. For example, all secondary network should be moved to _linux.go then.

So, I feel it does not help just to put NewController() to init_linux.go. For consideration of supporting Windows in future, it is also good make common code compliable on both Linux and Windows?

Reply:
So your suggestion is we modify the functions in secondaryNetwork module more common in this bugfix?

Reply:
Humm.. I was just suggesting moving NewController() to init.go, if it does not have any Linux dependency. Moving Initialize() and Restore() requires more code changes and probably more dummy funcs for Windows, and so not worthwhile.

Reply:
got it, I will update accordingly.